miio/miot_device.py

```diff
@@ -1,38 +1,117 @@
 import logging
+from enum import Enum
+from functools import partial
+from typing import Any, Union
 
+import click
+
+from .click_common import EnumType, LiteralParamType, command
 from .device import Device
+from .exceptions import DeviceException
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def _str2bool(x):
+    """Helper to convert string to boolean."""
+    return x.lower() in ("true", "1")
+
+
+# partial is required here for str2bool, see https://stackoverflow.com/a/40339397
+class MiotValueType(Enum):
+    Int = int
+    Float = float
+    Bool = partial(_str2bool)
+    Str = str
+
+
 class MiotDevice(Device):
     """Main class representing a MIoT device."""
 
-    def __init__(
-        self,
-        mapping: dict,
-        ip: str = None,
-        token: str = None,
-        start_id: int = 0,
-        debug: int = 0,
-        lazy_discover: bool = True,
-    ) -> None:
-        self.mapping = mapping
-        super().__init__(ip, token, start_id, debug, lazy_discover)
+    mapping = None
```

Comment:
Hello,
This seems to be a breaking change for ``MiotDevice``. We can no longer use 
```python
MiotDevice(ip=host, token=token, mapping=mapping)
```
to instantiate a MiotDevice with a given mapping (ha0y/xiaomi_miot_raw#79).
Is there a method to keep backward compatibility?

Reply:
Hi, sorry about that! You could assign the mapping directly after constructing the `MiotDevice` object to fix this for now:
```
dev = MiotDevice(...)
dev.mapping = my_mapping
```

Otherwise, feel free to open an issue to re-add the kwarg variant of this parameter back to the constructor!

Reply:
Thanks for your prompt reply!

Reply:
Unfortunatelly, this doesn't solve the problem, as this code doesn't work in version 0.5.4. `mapping` is required in 0.5.4 but unexpected in 0.5.5. I have opened an issue #982. Hope this will be fixed soon because it *is* a breaking change and may affect those who uses this library.

Reply:
Indeed, I hadn't thought about that, sorry... See my comment on the issue how I plan to rectify it.